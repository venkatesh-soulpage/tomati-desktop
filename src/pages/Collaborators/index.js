import React from "react";
import * as Action from "_actions";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

const Index = () => {
  return (
    <>
      <div className="p-4">
        <div>
          <div className="card px-4 py-4 mt-3 cr-p" style={{ borderRadius: 6 }}>
            <div className="d-flex align-items-center justify-content-between">
              <div>
                <h6 className="m-0 font-weight-bold">FirstName LastName</h6>
              </div>
              <div>
                <button className="btn btn-danger">Pending</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

function mapStateToProps(state) {
  return { auth: state.auth };
}

export default withRouter(connect(mapStateToProps)(Index));
