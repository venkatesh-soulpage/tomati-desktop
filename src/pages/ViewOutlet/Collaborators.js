import React from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import Loading from "components/Loading";

const Collaborators = (props) => {
  if (!props.outlet.outlet) {
    return (
      <div>
        <Loading textSecondary={true} />{" "}
      </div>
    );
  }

  return (
    <>
      {props?.outlet?.outlet?.collaborators?.length === 0 ? (
        <div className="text-center mt-5">
          you currently have no collaborators, add some
        </div>
      ) : (
        <div>
          {props?.outlet?.outlet?.collaborators?.map((collaborator, id) => {
            return (
              <div key={id} className="p-2 px-0 m-0">
                <div>
                  <div
                    className="card px-4 py-4 mt-1 cr-p"
                    style={{ borderRadius: 6 }}
                  >
                    <div className="d-flex align-items-center justify-content-between">
                      <div>
                        <h6 className="m-0 font-weight-bold">
                          {collaborator.email}
                        </h6>
                      </div>
                      <div>
                        <button className="btn btn-danger">
                          {collaborator.status}
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            );
          })}
        </div>
      )}
    </>
  );
};

function mapStateToProps(state) {
  return { outlet: state.outlet };
}

export default withRouter(connect(mapStateToProps)(Collaborators));
