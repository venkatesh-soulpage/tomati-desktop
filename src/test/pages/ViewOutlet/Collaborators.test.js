import React from "react";
import { render } from "@testing-library/react";
import Collaborators from "pages/ViewOutlet/Collaborators";
import { Router } from "react-router-dom";
import history from "utils/history";
import { Provider } from "react-redux";
import configureStore from "store";

describe("Collaborators page", () => {
  it("matches snapshot", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <Collaborators />
        </Router>
      </Provider>,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
