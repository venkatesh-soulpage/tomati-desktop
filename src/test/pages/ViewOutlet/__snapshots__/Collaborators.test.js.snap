// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collaborators page matches snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
